### PR TITLE
GitHub Actions for test

### PR DESCRIPTION
**Pull request by crazy-max**
Follow-up #451 

* `make test` GHA compatible
* Add local cache for GHA (binaries)
* Add badges in README
* Add godev workflow to "ping" pkg.go.dev for changes when a release is produced (see [Adding a package](https://go.dev/about)).
* Add codecov

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      - 'master'
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - 'master'
+
+env:
+  REPO_SLUG_ORIGIN: "moby/buildkit:master"
+  PLATFORMS: "linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64,linux/s390x,linux/ppc64le"
```
Comment (tonistiigi) [inline .github/workflows/build.yml:18]: looks unused

Comment (crazy-max) [reply .github/workflows/build.yml:16]: I'm a little ahead of schedule :sweat_smile:

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+comment: false
```
Comment (tonistiigi) [inline codecov.yml:1]: want to make same changes as https://github.com/moby/buildkit/pull/1878 for this?

File: hack/test [hunk 3]
```diff
@@ -3,51 +3,39 @@
...
+: ${TEST_COVERAGE=}
+
+importCacheFlags=""
+if [ -n "$cacheRefFrom" ]; then
+  if [ "$cacheType" = "local" ]; then
+    for ref in $cacheRefFrom; do
+      importCacheFlags="$importCacheFlags--cache-from=type=local,src=$ref "
+    done
+  else
+    importCacheFlags="--cache-from=type=registry,ref=$cacheRefFrom:integration-tests "
```
Comment (tonistiigi) [inline hack/test:15]: don't think image like `$cacheRefFrom:integration-tests` exists

File: hack/test [hunk 4]
```diff
@@ -3,51 +3,39 @@
...
-docker run --rm -v /tmp --volumes-from=$cacheVolume --privileged $iid go test ${TESTFLAGS:--v} ${TESTPKGS:-./...}
+buildxCmd build $importCacheFlags \
+  --target "integration-tests" \
+  --output "type=docker,name=$iid" \
+  $currentcontext
 
-if [ -n "$BUILDX_NOCACHE" ]; then
-  docker rm -v $cacheVolume
-fi
+cacheVolume=$(docker create -v /root/.cache -v /root/.cache/registry -v /go/pkg/mod alpine)
```
Comment (tonistiigi) [inline hack/test:36]: what are the registry references for. same for next line

Comment (tonistiigi) [inline hack/test:36]: previous code has capabilities for reusing the previous cache. without it separate volumes do not add much.

File: .github/workflows/build.yml (hunk 1 above)
Comment (crazy-max) [reply .github/workflows/build.yml]: @tonistiigi Done

File: hack/update-vendor [hunk 5]
```diff
@@ -1,45 +1,16 @@
 #!/usr/bin/env bash
 
 . $(dirname $0)/util
-set -eu -o pipefail -x
+set -eu
```
Comment (tonistiigi) [inline hack/update-vendor:4]: could we put non-test things to separate commit.

File: codecov.yml (hunk 2 above)
Comment (crazy-max) [reply codecov.yml:1]: I suspect that the issue on buildkit is related to testing on the windows environment (test-os job). I think we can leave it here and see what happens in the future.

File: hack/test (hunk 4 above)
Comment (crazy-max) [reply hack/test:36]: Good point, none actually

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**